core/src/main/java/org/apache/accumulo/core/fate/zookeeper/MetaFateStore.java

```diff
@@ -359,29 +365,50 @@ public void setStatus(TStatus status) {
 
     @Override
     public void delete() {
-      verifyReservedAndNotDeleted(true);
-
-      try {
-        zk.recursiveDelete(getTXPath(fateId), NodeMissingPolicy.SKIP);
-        this.deleted = true;
-      } catch (KeeperException | InterruptedException e) {
-        throw new IllegalStateException(e);
-      }
+      _delete(REQ_DELETE_STATUS);
     }
 
     @Override
     public void forceDelete() {
-      delete();
+      _delete(REQ_FORCE_DELETE_STATUS);
+    }
+
+    private void _delete(Set<TStatus> requiredStatus) {
+      verifyReservedAndNotDeleted(true);
+
+      // atomically check the txn status and delete the node
+      // retry until we either atomically delete the node or the txn status is disallowed
+      while (!this.deleted) {
+        Stat stat = new Stat();
+        FateData<T> fateData = getFateData(fateId, stat);
+        Preconditions.checkState(requiredStatus.contains(fateData.status),
+            "Tried to delete fate data for %s when the transaction status is %s", fateId,
+            fateData.status);
+        try {
+          zk.deleteStrict(getTXPath(fateId), stat.getVersion());
```

Comment:
changed to `deleteStrict` (from `recursiveDelete`) since we only expect the one node (and strict because we expect the same version). This also ensures that there is only one node as this will throw an exception if the dir is not empty